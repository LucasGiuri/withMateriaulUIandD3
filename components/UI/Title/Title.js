import PropTypes from 'prop-types';
import { Text } from './Title.styles';

const Title = ({ text, centered, isBold, isSmall }) => (
  <Text isBold={isBold} centered={centered} isSmall={isSmall}>{text}</Text>
);

Title.propTypes = {
  text: PropTypes.string.isRequired,
  centered: PropTypes.bool,
  isBold: PropTypes.bool,
  isSmall: PropTypes.bool
};

Title.defaultProps = {
  isSmall: false,
  isBold: false,
  centered: false
};

export default Title;