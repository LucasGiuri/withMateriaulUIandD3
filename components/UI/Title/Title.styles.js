import styled from 'styled-components';
import { PRIMARY } from '../../../public/styles/colors';

export const Text = styled.p`
  margin-bottom:  ${props => props.isSmall ? '0': '22px'};
  padding-left: 14px;
  padding-right: 14px;
  font-family: 'Roboto', sans-serif;
  color: ${PRIMARY};
  font-size: ${props => props.isSmall ? '16px': '22px'};
  text-align: ${props => props.centered ? 'center' : 'left'};
  font-weight: ${props => props.isBold ? 'bold' : 'normal'};
`;