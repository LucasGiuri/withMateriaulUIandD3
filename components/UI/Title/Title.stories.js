import React from 'react';
import { storiesOf } from '@storybook/react';
import Title from './Title';

const title = "Title";

storiesOf('Title')
  .add('Title default', () => (
    <Title text={title} />
  ))

  .add('Title centered', () => (
    <Title text={title} centered={true} />
  ))

  .add('Title isBod', () => (
    <Title text={title} isBold={true} />
  ))

  .add('Title isSmall', () => (
    <Title text={title} isSmall={true} />
  ))

  .add('Title centered and isBold', () => (
    <Title text={title} centered={true} isBold={true} />
  ))

  .add('Title centered and isSmall', () => (
    <Title text={title} centered={true} isSmall={true} />
  ))

  .add('Title isBold and isSmall', () => (
    <Title text={title} isBold={true} isSmall={true} />
  ))

  .add('Title centered, isBold and isSmall', () => (
    <Title text={title} centered={true} isBold={true} isSmall={true} />
  ))