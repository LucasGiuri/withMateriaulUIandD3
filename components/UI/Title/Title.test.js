import React from 'react';
import renderer from 'react-test-renderer';
import Title from './Title';
import 'jest-styled-components';

describe('Title', () => {
  it('Title basic render snapshoot', () => {
    const component = renderer.create(<Title text="text" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('Title centered render snapshoot', () => {
    const component = renderer.create(<Title text="text" centered={true} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('text-align', 'center');

  })

  it('Title isBold render snapshoot', () => {
    const component = renderer.create(<Title text="text" isBold={true} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('font-weight', 'bold');
  })

  it('Title isSmall render snapshoot', () => {
    const component = renderer.create(<Title text="text" isSmall={true} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('font-size', '16px');
  })

  it('Title isBold and centered render snapshoot', () => {
    const component = renderer.create(<Title text="text" centered={true} isBold={true} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('text-align', 'center');
    expect(tree).toHaveStyleRule('font-weight', 'bold');
  })

  it('Title isSmall and centered render snapshoot', () => {
    const component = renderer.create(<Title text="text" centered={true} isSmall={true} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('text-align', 'center');
    expect(tree).toHaveStyleRule('font-size', '16px');
  })

  it('Title isSmall and isBold render snapshoot', () => {
    const component = renderer.create(<Title text="text" isSmall={true} isBold={true} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('font-size', '16px');
    expect(tree).toHaveStyleRule('font-weight', 'bold');
  })

  it('Title isBold, isSmall and centered render snapshoot', () => {
    const component = renderer.create(<Title text="text" isSmall={true} centered={true} isBold={true} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('text-align', 'center');
    expect(tree).toHaveStyleRule('font-weight', 'bold');
    expect(tree).toHaveStyleRule('font-size', '16px');
  })
})
