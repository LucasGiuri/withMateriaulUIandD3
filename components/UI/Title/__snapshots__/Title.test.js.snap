// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Title basic render snapshoot 1`] = `
.c0 {
  margin-bottom: 22px;
  padding-left: 14px;
  padding-right: 14px;
  font-family: 'Roboto',sans-serif;
  color: #DC3545;
  font-size: 22px;
  text-align: left;
  font-weight: normal;
}

<p
  className="c0"
>
  text
</p>
`;

exports[`Title Title centered render snapshoot 1`] = `
.c0 {
  margin-bottom: 22px;
  padding-left: 14px;
  padding-right: 14px;
  font-family: 'Roboto',sans-serif;
  color: #DC3545;
  font-size: 22px;
  text-align: center;
  font-weight: normal;
}

<p
  className="c0"
>
  text
</p>
`;

exports[`Title Title isBold and centered render snapshoot 1`] = `
.c0 {
  margin-bottom: 22px;
  padding-left: 14px;
  padding-right: 14px;
  font-family: 'Roboto',sans-serif;
  color: #DC3545;
  font-size: 22px;
  text-align: center;
  font-weight: bold;
}

<p
  className="c0"
>
  text
</p>
`;

exports[`Title Title isBold render snapshoot 1`] = `
.c0 {
  margin-bottom: 22px;
  padding-left: 14px;
  padding-right: 14px;
  font-family: 'Roboto',sans-serif;
  color: #DC3545;
  font-size: 22px;
  text-align: left;
  font-weight: bold;
}

<p
  className="c0"
>
  text
</p>
`;

exports[`Title Title isBold, isSmall and centered render snapshoot 1`] = `
.c0 {
  margin-bottom: 0;
  padding-left: 14px;
  padding-right: 14px;
  font-family: 'Roboto',sans-serif;
  color: #DC3545;
  font-size: 16px;
  text-align: center;
  font-weight: bold;
}

<p
  className="c0"
>
  text
</p>
`;

exports[`Title Title isSmall and centered render snapshoot 1`] = `
.c0 {
  margin-bottom: 0;
  padding-left: 14px;
  padding-right: 14px;
  font-family: 'Roboto',sans-serif;
  color: #DC3545;
  font-size: 16px;
  text-align: center;
  font-weight: normal;
}

<p
  className="c0"
>
  text
</p>
`;

exports[`Title Title isSmall and isBold render snapshoot 1`] = `
.c0 {
  margin-bottom: 0;
  padding-left: 14px;
  padding-right: 14px;
  font-family: 'Roboto',sans-serif;
  color: #DC3545;
  font-size: 16px;
  text-align: left;
  font-weight: bold;
}

<p
  className="c0"
>
  text
</p>
`;

exports[`Title Title isSmall render snapshoot 1`] = `
.c0 {
  margin-bottom: 0;
  padding-left: 14px;
  padding-right: 14px;
  font-family: 'Roboto',sans-serif;
  color: #DC3545;
  font-size: 16px;
  text-align: left;
  font-weight: normal;
}

<p
  className="c0"
>
  text
</p>
`;
